core/src/main/scala/org/bykn/bosatsu/codegen/python/PythonGen.scala

```diff
@@ -1878,17 +1878,8 @@ object PythonGen {
             // there is no need to
             loop(in, slotName)
           case Literal(lit) => Env.pure(Code.litToExpr(lit))
-          case If(cond, thenExpr, elseExpr) =>
-            def combine(expr: Expr): (List[(BoolExpr, Expr)], Expr) =
-              expr match {
-                case If(c1, t1, e1) =>
-                  val (ifs, e2) = combine(e1)
-                  (ifs :+ ((c1, t1)), e2)
```

Comment:
note the order is wrong here: we are pushing `c1, t1` on to the end of `ifs` but that came from expanding `e1` so it should come before `e1` since it was before originally.